app/scrapers/courses_scraper.rb

```diff
@@ -53,45 +53,117 @@ def utf_safe text
+    coreq = match ? match[1] : nil
+
+    match = /(?:Restricted to)|(?:Restriction:) ([^.]+\.)/.match(text)
+    text = match ? text.gsub(match[0], '') : text
+    restrictions = match ? match[1] : nil
+
+    match = /Credit (?:(?:only )|(?:will be ))?granted for(?: one of the following)?:? ([^.]+\.)/.match(text)
+    text = match ? text.gsub(match[0], '') : text
+    credit_granted_for = match ? match[1] : nil
+
+    match = /Also offered as:? ([^.]+\.)/.match(text)
+    text = match ? text.gsub(match[0], '') : text
+    also_offered_as = match ? match[1] : nil
+
+
+    match = /Formerly:? ([^.]+\.)/.match(text)
+    text = match ? text.gsub(match[0], '') : text
+    formerly = match ? match[1] : nil
+
+    match = /Additional information: ([^.]+\.)/.match(text)
+    text = match ? text.gsub(match[0], '') : text
+    additional_info = match ? match[1] : nil
+
+    # if approved-course-texts held relationships, use 2nd child as description and leftover text as "additional info"
+    if approved.css('> div').length > 0 then
+
+      description = utf_safe approved.css('> div:last-child').text.strip.gsub(/\t|(\r\n)/, '')
+      additional_info = additional_info ? additional_info += ' '+text : text
+      additional_info = additional_info && additional_info.strip.empty? ? nil : additional_info.strip
+
+    elsif other.css('> div').length > 0 then
+      description = text.strip.empty? ? nil : text.strip
+    end
+
     relationships = {
-      coreqs: /Corequisite: ([^.]+)/.match(description).to_a[2],
-      prereqs: /Prerequisite: ([^.]+)/.match(description).to_a[2],
-      restrictions: /Restriction: ([^.]+)/.match(description).to_a[2],
-      restricted_to: /Restricted to ([^.]+)/.match(description).to_a[2],
-      credit_only_granted_for: /Credit only granted for:([^.]+)/.match(description).to_a[2],
-      credit_granted_for: /Credit granted for([^.]+)/.match(description).to_a[2],
-      formerly: /Formerly:([^.]+)/.match(description).to_a[2],
-      also_offered_as: /Also offered as([^.]+)/.match(description).to_a[2]
-    } 
-
-    course_array << {
-      course_id: course.search('div.course-id').text,
-      name: course.css('span.course-title').first.content,
+      prereqs: prereq,
+      coreqs: coreq,
+      restrictions: restrictions,
+      credit_granted_for: credit_granted_for,
+      also_offered_as: also_offered_as,
+      formerly: formerly,
+      additional_info: additional_info 
+    }
+
+    courses << {
+      course_id: course_id,
+      name: course_title,
       dept_id: dept_id,
       department: department,
       semester: semester,
       credits: course.css('span.course-min-credits').first.content,
-      grading_method: course.at_css('span.grading-method abbr') ? course.at_css('span.grading-method abbr').attr('title').split(', ') : [],
+      grading_method: course.at_css('span.grading-method abbr') ? 
+              course.at_css('span.grading-method abbr').attr('title').split(', ') : [],
```

Comment:
nit: safe navigation operator (`&`) could also be used here, e.g.: `course.at_css('span.grading-method abbr')&.attr('title')&.split(', ') || []`
Where anything that evals to nil will stop at the first ampersand and the whole expression becomes nil. Also using `nil || []` evaluates to `[]`